### PR TITLE
add Kuikae

### PR DESCRIPTION
**Pull request by ktc312**
# Description

Check for Kuikae (喰い替え) after Chii or Pon
Related issue: #99 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the tests and make sure they pass
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: mahjong/player.py [hunk 1]
```diff
@@ -152,16 +152,29 @@ def action_with_new_tile(
...
+        """
         # add tmp_huro to kabe
         self.kabe.append(self.tmp_huro)
         if naki != Naki.ANKAN:
             self.menzenchin = False
+
+        # check Kuikae 喰い替え
+        kuikae_tiles = []
+        if naki == Naki.CHII:
+            kuikae_tiles.append(self.tmp_huro.tiles)
```
Comment (ktc312) [inline mahjong/player.py:168]: 改成尾巴的牌，和吃進來的牌

File: mahjong/player.py [hunk 2]
```diff
@@ -225,12 +238,22 @@ def get_input(
...
+    ) -> Tile:
         """Add in the newly drawn tile and discard a tile
         """
         hand_tiles = convert_hand(self.hand)
         if new_tile:
             hand_tiles.append(new_tile)
+        if kuikae_tiles:
+            # not allowed to choose from this list
+            # TODO: now just hide from hand_representation,
+            # need to change to something else if UI changes
```
Comment (ktc312) [inline mahjong/player.py:254]: edge case 手牌剩 3 3 4 5，吃 6 ，會花生什麼事？

Comment (ktc312) [reply mahjong/player.py:261]: 不能吃

Comment (ktc312) [reply mahjong/player.py:261]: https://github.com/rasca0027/Mahjong4RL/issues/103

**Review by hoaaoh — APPROVED** [review]

**Review by HappyBall — APPROVED** [review]

**Pull request merged**